backend/functions/lib/adapters/types/authorization.ts

```diff
@@ -0,0 +1,55 @@
+import { CamsUser } from '../../../../../common/src/cams/session';
+
+export type AuthorizationConfig = {
+  provider: string;
+  issuer: string;
+  audience: string;
+  userInfoUri: string;
+};
+
+export interface OpenIdConnectGateway {
+  verifyToken: (accessToken: string) => Promise<Jwt>;
+  getUser: (accessToken: string) => Promise<CamsUser>;
+}
+
+export type Algorithm =
+  | 'HS256'
+  | 'HS384'
+  | 'HS512'
+  | 'RS256'
+  | 'RS384'
+  | 'RS512'
+  | 'ES256'
+  | 'ES384'
+  | 'ES512'
+  | 'none';
+
+export type JwtHeader = {
+  alg: Algorithm;
+  typ: string;
+  kid?: string;
+  jku?: string;
+  x5u?: string;
+  x5t?: string;
+};
+
+export type JwtClaims = {
+  iss: string;
+  sub: string;
+  aud: string;
+  exp: number;
+  nbf?: number;
+  iat?: number;
+  jti?: string;
+  nonce?: string;
+  scp?: string[];
+  [key: string]: unknown;
+};
+
+export type Jwt = {
```

Comment:
I left a comment about this on another file, but I think we should not use this type outside of the Okta gateway. The JwtHeader and JwtClaims also have some properties that are not included in [the JWT spec](https://datatracker.ietf.org/doc/html/rfc7519#section-5), but are presumably implementation details meaningful only in the context of the Okta library.

Reply:
That much makes sense. Do the Okta customizations fall under the `[key: string]: unknown` catch all in our copy of the type lib?

Reply:
I haven't done a gap yet. I just took a cursory look.